doc/radosgw/pubsub-module.rst

```diff
@@ -0,0 +1,250 @@
+=========================
+PubSub Sync Module
+=========================
+
+.. versionadded:: Nautilus
+
+This sync module provides publish and subscribe mechanism for the object store modification
+events. Events are published into defined topics. Topics can be subscribed to, and events
+can be pulled from them. Events need to be acked. Also, events will expire and disappear
+after a period of time. A basic push notification mechanism exists too, but it is not
+reliable.
+
+A user can create different topics. A topic entity is defined by its user and its name. A
+user can only manage its own topics, and can only subscribe to events published by buckets
+it owns.
+
+In order to publish events for specific bucket a notification needs to be created. A
+notification can be created only on subset of event types, or for all event types (default).
+There can be multiple notifications for any specific topic.
+
+A subscription to a topic can also be defined. There can be multiple subscriptions for any
+specific topic.
+
+New REST api was define to provide configuration and control interfaces for the pubsub
```

Comment:
Some syntax edit needed here.  Could be "A new REST api has been defined" or maybe "New REST api methods are defined"?

Reply:
@mattbenjamin had to force push due to rebase screwup, but took care of both your comments (see https://github.com/ceph/ceph/pull/23298/commits/b143295ee7eb3e379b078342b494c91892bcf98d)